tmt/steps/provision/connect.py

```diff
@@ -110,8 +111,31 @@ def go(self):
             data['key'] = key
 
         # And finally create the guest
-        self._guest = tmt.Guest(data, name=self.name, parent=self.step)
+        self._guest = GuestConnect(data, name=self.name, parent=self.step)
 
     def guest(self):
         """ Return the provisioned guest """
         return self._guest
+
+
+class GuestConnect(tmt.Guest):
+    """
+    Guest connection class.
+    """
+
+    def reboot(self, hard=False, machine_name=None):
+        super.reboot(hard, name)
+        conn_timeout = 60 * 4
+        self.debug("Wait for an ssh connection to the machine.")
+        for i in range(1, conn_timeout):
+            try:
+                self.execute('whoami')
+                break
+            except tmt.utils.RunError:
+                self.debug('Failed to connect to the machine, retrying.')
+            time.sleep(1)
+
+        if i == conn_timeout:
+            self.debug("Failed to boot the machine, removing it.")
```

Comment:
Is it really removing it? If so, what's removing the machine?

Reply:
Fixed.